code/modules/clothing/under/striketeam.dm

```diff
@@ -0,0 +1,134 @@
+
+/obj/item/clothing/under/deathsquad/New()
+	..()
+	deathsquad_uniforms += src
+
+/obj/item/clothing/under/deathsquad/Destroy()
+	deathsquad_uniforms -= src
+	..()
+
+/obj/item/clothing/under/deathsquad/ui_action_click()
+	togglemap()
+
+/obj/item/clothing/under/deathsquad/process()
+	update_holomap()
+
+/obj/item/clothing/under/proc/togglemap()
+	if(usr.isUnconscious())
+		return
+
+	if(!ishuman(usr))
+		to_chat(usr, "<span class='warning'>Only humanoids can wear this suit</span>")
+		return
+
+	var/mob/living/carbon/human/H = usr
+
+	if(H.get_item_by_slot(slot_w_uniform) != src)
+		to_chat(H, "<span class='warning'>You need to wear the suit first</span>")
+		return
+
+	if(src.holomap_activated)
+		deactivate_holomap()
+		to_chat(H, "<span class='notice'>You disable the holomap.</span>")
+	else
+		holomap_activated = 1
+		activator = H
+		processing_objects.Add(src)
+		process()
+		to_chat(H, "<span class='notice'>You enable the holomap.</span>")
+
+#define HOLOMAP_ERROR	0
+#define HOLOMAP_YOU		1
+#define HOLOMAP_OTHER	2
+#define HOLOMAP_DEAD	3
+
+/obj/item/clothing/under/proc/update_holomap()
+	var/turf/T = get_turf(src)
+	if(!T)//nullspace begone!
+		return
+
+	if((activator.get_item_by_slot(slot_w_uniform) != src) || (!activator.client) || (holoMiniMaps[T.z] == null))
+		deactivate_holomap()
+		return
+
+	activator.client.images -= holomap_images
+
+	holomap_images.len = 0
+
+	var/image/bgmap
+
+	if(T.z == map.zCentcomm)
+		bgmap = image(centcommMiniMaps[HOLOMAP_FILTER_DEATHSQUAD])
+	else
+		bgmap = image(holoMiniMaps[T.z])
+	bgmap.pixel_x = -1*T.x + 7*WORLD_ICON_SIZE + 16*(WORLD_ICON_SIZE/32)
+	bgmap.pixel_y = -1*T.y + 7*WORLD_ICON_SIZE + 17*(WORLD_ICON_SIZE/32)
+	bgmap.plane = HUD_PLANE
+	bgmap.layer = HUD_BASE_LAYER
+	bgmap.color = "#0B74B4"
+	bgmap.loc = activator.hud_used.holomap_obj
+	holomap_images += bgmap
+
+	for(var/obj/item/clothing/under/deathsquad/D in deathsquad_uniforms)
+		var/mob_indicator = HOLOMAP_ERROR
+		var/turf/TD = get_turf(D)
+		if(D == src)
+			mob_indicator = HOLOMAP_YOU
+		else if((TD.z == T.z) && ishuman(D.loc))
+			var/mob/living/carbon/human/H = D.loc
+			if(H.get_item_by_slot(slot_w_uniform) == D)
+				if(H.isDead())
+					mob_indicator = HOLOMAP_DEAD
+				else
+					mob_indicator = HOLOMAP_OTHER
+			else
+				continue
+
+		if(mob_indicator != HOLOMAP_ERROR)
+
+			var/holomap_marker = "deathsquad_\ref[src]_\ref[D]"
+
+			if(!(holomap_marker in holomap_cache))
+				holomap_cache[holomap_marker] = image('icons/12x12.dmi',"ds[mob_indicator]")
+
+			var/image/I = holomap_cache[holomap_marker]
+			I.pixel_x = TD.x - T.x + 7*WORLD_ICON_SIZE + 8*(WORLD_ICON_SIZE/32)
+			I.pixel_y = TD.y - T.y + 7*WORLD_ICON_SIZE + 9*(WORLD_ICON_SIZE/32)
+			I.plane = HUD_PLANE
+			if(mob_indicator == HOLOMAP_YOU)
+				I.layer = HUD_ABOVE_ITEM_LAYER
+			else
+				I.layer = HUD_ITEM_LAYER
+			I.loc = activator.hud_used.holomap_obj
+			animate(I,alpha = 255, time = 13, loop = -1, easing = SINE_EASING)
```

Comment:
The animations are stopped and the images deleted in the uniform's Destroy() proc. In the meantime they remain cached.